python_dwd/additionals/functions.py

```diff
@@ -178,9 +178,15 @@ def check_parameters(parameter: Parameter,
     check = TIME_RESOLUTION_PARAMETER_MAPPING.get(time_resolution, [[], []])
 
     if parameter not in check[0] or period_type not in check[1]:
-        raise NameError(
+        raise TypeError(
             f"Combination of time_resolution={time_resolution.value},parameter={parameter.value} "
             f"and period_type={period_type.value} not available.Possible parameters are: "
             f"{TIME_RESOLUTION_PARAMETER_MAPPING}.")
 
     return None
+
+
+def find_all_matchstrings_in_string(string: str,
```

Comment:
Do you mean parameters? -> find_all_parameters_in_string

Reply:
No, this function is used to proove that all strings like "pordukt", ".txt" or whatever is in matchstrings, is found in string. This function is used to identify a specific filetype like stationdata. Originally the search for this file was a bit messy as it had a nested list comprehension.